services_app/src/main/java/org/opendatakit/services/sync/actions/fragments/SyncFragment.java

```diff
@@ -53,566 +56,606 @@
+                                                @Override
+                                                public void run() {
+                                                    updateInterface();
+                                                }
+                                            }, 100);
+                                        }
+                                    }
+                                });
+                                return;
+                            }
+                        } else {
+                            // The service is not bound yet so now we need to try again
+                            handler.postDelayed(new Runnable() {
+                                @Override
+                                public void run() {
+                                    updateInterface();
+                                }
+                            }, 100);
                         }
-                      }, 100);
                     }
-                  }
                 });
-                return;
-              }
-            } else {
-              // The service is not bound yet so now we need to try again
-              handler.postDelayed(new Runnable() {
-                @Override public void run() {
-                  updateInterface();
-                }
-              }, 100);
-            }
-          }
-        });
-  }
-
-  void syncCompletedAction(IOdkSyncServiceInterface syncServiceInterface) throws RemoteException {
-    removeAnySyncNotification();
-    boolean completed = syncServiceInterface.clearAppSynchronizer(getAppName());
-    if (!completed) {
-      throw new IllegalStateException("Could not remove AppSynchronizer for " + getAppName());
     }
-    perhapsEnableButtons();
-    updateInterface();
-  }
-
-  /**
-   * Hooked to sync_reset_server_button's onClick in sync_launch_fragment.xml
-   */
-  public void onClickResetServer(View v) {
-    WebLogger.getLogger(getAppName()).d(TAG, "[" + getId() + "] [onClickResetServer]");
-    // ask whether to sync app files and table-level files
-
-    if (areCredentialsConfigured(false)) {
-      // show warning message
-      AlertDialog.Builder msg = buildOkMessage(getString(R.string.sync_confirm_reset_app_server),
-          getString(R.string.sync_reset_app_server_warning));
-
-      msg.setPositiveButton(getString(R.string.sync_reset), new DialogInterface.OnClickListener() {
-        @Override public void onClick(DialogInterface dialog, int which) {
-          WebLogger.getLogger(getAppName()).d(TAG,
-              "[" + getId() + "] [onClickResetServer] timestamp: " + System.currentTimeMillis());
-          disableButtons();
-          syncAction = SyncActions.RESET_SERVER;
-          prepareForSyncAction();
+
+    void syncCompletedAction(IOdkSyncServiceInterface syncServiceInterface) throws RemoteException {
+        removeAnySyncNotification();
+        saveLastSyncInfo();
```

Comment:
This would update the timestamp even if the last sync failed. Take a look at the `updateInterface` method to see how to check the sync outcome.